### PR TITLE
Fixes chemtanks for good this time

### PR DESCRIPTION
**Pull request by GuyonBroadway**
For some reason the ordering on the if statements was wrong... whoops.

:cl: GuyonBroadway
fix: Chem tanks now ACTUALLY work this time (honest!)
/:cl:

### CODE REVIEW

**Comment by PopNotes** [comment]
tyring

**Comment by GuyonBroadway** [comment]
yer tyring ym patinece

**Comment by tortellinitony** [comment]
Jesus Christ this file is a mess

File: hippiestation/code/modules/reagents/reagent_containers/glass.dm [hunk 1]
```diff
@@ -16,8 +16,12 @@
 			var/trans = reagents.trans_to(target, amount_per_transfer_from_this)
 			to_chat(user, "<span class='notice'>You transfer [trans] unit\s of the solution to [target].</span>")
 
+		if(!target.is_open_container() && target.reagents.total_volume != 0)
```
Comment (tortellinitony) [inline hippiestation/code/modules/reagents/reagent_containers/glass.dm]: From the looks of it, this doesn't check if `src` is full before adding reagents to it...

Just have this:

```
else
	if(reagents.total_volume >= reagents.maximum_volume)
		to_chat(user, "<span class='notice'>[src] is full.</span>")
		return
	if(!target.reagents.total_volume)
		to_chat(user, "<span class='warning'>[target] is empty!</span>")
		return
	else
		var/trans = target.reagents.trans_to(src, amount_per_transfer_from_this)
		to_chat(user, "<span class='notice'>You fill [src] with [trans] unit\s of the contents of [target].</span>")
		return
```

**Review by tortellinitony — DISMISSED** [review]

**Comment by KayeArray** [comment]
ahhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

File: hippiestation/code/modules/reagents/reagent_containers/glass.dm (hunk 1 above)
Comment (GuyonBroadway) [reply hippiestation/code/modules/reagents/reagent_containers/glass.dm]: this just makes it get stuck in an infinite loop where it puts a chem into the tank then takes it right out again.

Comment (tortellinitony) [reply hippiestation/code/modules/reagents/reagent_containers/glass.dm]: what

Comment (tortellinitony) [reply hippiestation/code/modules/reagents/reagent_containers/glass.dm]: this isn't a loop how could it be an infinite loop if it isn't even a loop

Comment (GuyonBroadway) [reply hippiestation/code/modules/reagents/reagent_containers/glass.dm]: sorry not infinite but it does add a chem to the beaker and then take it straight back out.

Comment (tortellinitony) [reply hippiestation/code/modules/reagents/reagent_containers/glass.dm]: how

Comment (GuyonBroadway) [reply hippiestation/code/modules/reagents/reagent_containers/glass.dm]: Hang on I think I got it... had the silly tabs in the wrong place.

**Review by j-mie — APPROVED** [review]

**Pull request merged**